compiler/rustc_lint_defs/src/lib.rs

```diff
@@ -39,6 +39,39 @@ macro_rules! pluralize {
     };
 }
 
+/// Grammatical tool for displaying messages to end users in a nice form.
+///
+/// Returns "an" if the given string starts with a vowel, and "a" otherwise.
+pub fn a_or_an(s: &str) -> &'static str {
```

Comment:
Open to suggestions on placement of something like this, put here because it seemed related to the `pluralize` macro initially but this feels like the wrong place at the same time.